pandas/core/internals/blocks.py

```diff
@@ -7,7 +7,8 @@
 
 import numpy as np
 
-from pandas._libs import NaT, Timestamp, lib, tslib, writers
+from pandas._libs import NaT, lib, tslib, writers
+from pandas._libs.index import convert_scalar
```

Comment:
followup: convert_scalar should be in pandas.core.dtypes.cast (you can just call the cython version), but import / expose from there.

also the cython version is in a very weird place (index.pyx).